server/portal/apps/workspace/api/views.py

```diff
@@ -90,6 +91,16 @@ def get(self, request, *args, **kwargs):
         if app_id:
             METRICS.debug("user:{} is requesting app id:{}".format(request.user.username, app_id))
             data = _get_app(app_id, request.user)
+            
+            # check if default system needs keys pushed
+            default_sys = UserSystemsManager(
+                request.user,
+                settings.PORTAL_DATA_DEPOT_LOCAL_STORAGE_SYSTEM_DEFAULT
+            )
+            storage_sys = StorageSystem(agave, default_sys.get_system_id())
+            success, result = storage_sys.test()
+            data['systemHasKeys'] = success
+            data['pushKeysSystem'] = storage_sys.to_dict()
```

Comment:
In general, I'm opposed to the idea of injecting information into the app definition that is not natively a part of the app definition, for the following reasons:
- This could make it difficult to refactor when we try to go to TAPIS v3
- Fixtures become complicated, since you can't use a single client-side app fixture without injecting this other information (as opposed to building the response from the app fixture and a storage system fixture)

I propose refactoring this view to nest the definition, other flags and storage system, so the response would to /api/workspace/apps/?app_id={app} to return:

```
{
  'definition': {},
  'systemHasKeys': false,
  'pushKeysSystem': {}
}

Reply:
Oh okay, I looked at `_get_app` and thought the response `data` for the `AppsView` was an object containing the app definition. If `data` _is_ the app definition, then I might have to sort out a few potential issues in the apps/jobs sagas.

Reply:
data is currently the app definition with extra stuff. but we're jamming so much extra things (license info, required system defs)  into it i think we should separate them from the definition, rather than injecting them into the definition.

Reply:
I'm working on the changes, and will put them up in a feature branch pointed towards this one.

Reply:
Being followed up in FP-630 ( https://github.com/TACC/Core-Portal/pull/381 )